### ISSUE TITLE
Empty popup when right-clicking favorites

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot#60921]
### Godot version

5dae0cb

### System information

Windows 10 x64

### Issue description

When you right-click any file in filesystem dock and then right-click favorites, you end up with empty popup.
![godot windows tools 64_kV431JxXxN](https://user-images.githubusercontent.com/2223172/167617119-79c027ca-e8e8-4ea6-87fa-d6be04324fb8.gif)
Normally right-clicking favorites doesn't do anything, so likely what happens is that you have a popup visible and then it gets 
cleared.

### Steps to reproduce

1. Right-click any file in filesystem dock
2. Right-click favorites item

**Comment by akien-mga** [comment]
I can't reproduce it on Linux X11, it might be Windows specific. Seems to happen both on Windows 10 for @KoBeWi and Windows 11 for @lufog.

**Comment by KoBeWi** [comment]
Likely caused by #59287
The popup should just disappear.

**Comment by derammo** [comment]
I'm close to fixing this without disabling right clicks, I believe.

**Comment by derammo** [comment]
Update:

This seems like a sequence problem, with a deferred operation triggering itself in the popup code.  The first popup gets closed twice, but in the meantime, it has been reused and moved to the new location.  Like so (not all frames shown):

```
OS_Windows::run
    process messages
        MouseProc right click
            _send_window_event WINDOW_EVENT_CLOSE_REQUEST
                Window::_propagate_window_notification NOTIFICATION_WM_CLOSE_REQUEST
                    PopupMenu::_notification
                        Popup::_notification
                            _close_pressed
                                call_deferred this->Window::hide
    Main::iteration
        Scene_tree::physics_process
            flush dispatches Window::hide
                set_visible
                    _clear_window
                        delete_sub_window
                            popup_close
                                _send_window_event WINDOW_EVENT_CLOSE_REQUEST
                                    Window::_propagate_window_notification NOTIFICATION_WM_CLOSE_REQUEST
                                        PopupMenu::_notification
                                            Popup::_notification
                                                _close_pressed
                                                    call_deferred this->Window::hide

OS_Windows::run
    process messages
        WndProc right click
            parse_input_event
                buffered_events.push_back
    Main::iteration
        Input::flush_buffered_events
            long way through the gui ...
                PopupMenu::popup
                    Window::popup
                        set_visible (true)
                            window set visible in new location
        Scene_tree::physics_process
            flush dispatches Window::hide from previous sequence
                set_visible (false)
                    window was reused in the time we were async, close it again
```

Now to find a clean fix.  The challenge is that there are multiple ways that popups are called, sometimes the window is set to hidden from code, other times the MouseProc calls directly into it.  With the current structure, I haven't found a clean way to prevent the asynchronous recursion.   The popup code doesn't really know in which context it is used, whether it is responsible for scheduling the Window::hide.  I am trying to find a fix for that without restructuring.

**Comment by derammo** [comment]
> Likely caused by 59287 The popup should just disappear.

@KoBeWi yes, all my testing is with that PR reverted.  Are you sure the popup should just close and not be shown in the new location in this scenario?  Right now, that is what the code is trying to do, the right click in the new location opens the menu there and the extra mouse handler bug unfortunately closes the window immediately.

**Comment by derammo** [comment]
I have a fix, just waiting on PR in the same area to go in first for clean rebase.
